### ISSUE TITLE
`exported` array setting in `.pants.boostrap` parsed differently by `pants` than by `./pants`

### ISSUE THREAD

**Issue by danxmoran** [pantsbuild/scie-pants#107]
In our `.pants.bootstrap`, we have:
```
export PANTS_DOCKER_TOOLS="+['aws-oidc', 'open']"
```
(It's a conditional `export`, otherwise we'd put it in `pants.toml`)

When running `./pants`, this `export` works to append the 2 values to the end of the config specified in `pants.toml`. When running the `pants` binary, the setting instead seems to overwrite the existing value with the escaped string contents of the env var, resulting in the error:
```
pants.core.util_rules.system_binaries.BinaryNotFoundError: Cannot find `'+['\''aws-oidc'\'', '\''open'\'']'` on `['/System/Cryptexes/App/usr/bin', '/Users/dan.moran/.asdf/shims', '/Users/dan.moran/.cargo/bin', '/Users/dan.moran/.go/bin', '/Users/dan.moran/bat-extras/bin', '/bin', '/opt/homebrew/bin', '/sbin', '/usr/bin', '/usr/local/MacGPG2/bin', '/usr/local/bin', '/usr/local/go/bin', '/usr/local/opt/asdf/libexec/bin', '/usr/local/opt/python3/libexec/bin', '/usr/local/sbin', '/usr/sbin']`. Please ensure that it is installed so that Pants can use docker.
```

**Comment by jsirois** [comment]
Both of these issues can be solved by re-execing through bash instead of sandboxing a bash source. I'm still working through a test detail, but this is solved.

**Comment by kaos** [comment]
> I'll add to this that this `.pants.bootstrap` content doesn't work the same either:
> 
> ```shell
> export COLUMNS=$(tput cols)
> export LINES=$(tput lines)
> ```

Ah, this is broken in a slightly different way from what I initially thought. It does export and propagate, only it is reporting the wrong numbers.. i.e. it doesn't solve the issue when used with `scie-pants`, where as when used with `./pants` it does.

```bash
# cat .pants.bootstrap
export COLUMNS=$(tput cols)
export LINES=$(tput lines)

echo "COLS: $COLUMNS, LINES: $LINES"
```

```
$ ./pants global --help-advanced
COLS: 241, LINES: 69
[...]

$ pants global --help-advanced
COLS: 80, LINES: 24
[...]
```

**Comment by jsirois** [comment]
Yeah, exactly. The fix fixes this use case anyhow and there is an IT for it now.

**Comment by jsirois** [comment]
This fix is now live in the 0.4.2 release.

**Comment by kaos** [comment]
> This fix is now live in the 0.4.2 release.

Confirmed working for my `tput` use-case 👍🏽
